File: keanu-research/PocketKeanu/src/main/java/com/research/pocketKeanu/distributions/ContinuousConditionalProb.java
package com.research.pocketKeanu.distributions;

import com.research.pocketKeanu.autoDiff.DifferentiableDouble;

public interface ContinuousConditionalProb {
    DifferentiableDouble logProb(DifferentiableDouble [] randVars, DifferentiableDouble [] givens);
}

File: keanu-research/PocketKeanu/src/main/java/com/research/pocketKeanu/distributions/ContinuousConditionalProb.kt
package com.research.pocketKeanu.distributions

import com.research.pocketKeanu.abstractTypes.DoubleLike
import com.research.pocketKeanu.randomFactory.DoubleLikeRandomFactory

interface ContinuousConditionalProb {
    fun <T : DoubleLike<T>> logProb(randVars: DoubleLikeRandomFactory<T>, givens: List<T>): T
}
